botnim/query.py

```diff
@@ -111,20 +112,115 @@ def get_index_mapping(self) -> Dict:
             logger.error(f"Failed to get index mapping: {str(e)}")
             raise
 
-def run_query(query_text: str, environment: str, bot_name: str, context_name: str, num_results: int = 7) -> List[SearchResult]:
+    def get_default_num_results(self, context_name: str) -> int:
+        """
+        Get the default number of results for a context
+        
+        Args:
+            context_name (str): Name of the context
+            
+        Returns:
+            int: Default number of results
+        """
+        # Find the specific context configuration
+        context_config = next(
+            (ctx for ctx in self.config.get('context', []) if ctx['name'] == context_name),
+            {}
+        )
+        
+        # Get default num_results from context config or use a reasonable default
+        return context_config.get('default_num_results', 7)
+
+    def query(self, query_text: str, bot_name: str = None, context_name: str = None, num_results: int = None) -> List[SearchResult]:
+        """
+        Query the vector store with the given text
+        
+        Args:
+            query_text (str): The text to search for
+            bot_name (str, optional): Name of the bot to use (defaults to self.bot_name)
+            context_name (str, optional): Name of the context to search in (defaults to self.context_name)
+            num_results (int, optional): Number of results to return, or None to use context default
+        
+        Returns:
+            List[SearchResult]: List of search results
+        """
+        # Use instance values if not provided
+        bot_name = bot_name or self.bot_name
```

Comment:
Why is bot_name needed here?

Reply:
don't overwrite bot name and context name. never.

Reply:
fixed